impeller/renderer/command.cc

```diff
@@ -120,4 +121,41 @@ bool Command::BindResource(ShaderStage stage,
   return false;
 }
 
+bool Command::BindResource(ShaderStage stage,
+                           DescriptorType type,
+                           const SampledImageSlot& slot,
+                           const ShaderMetadata& metadata,
+                           std::shared_ptr<const Texture> texture,
+                           const std::unique_ptr<const Sampler>& sampler) {
+  if (!sampler) {
+    return false;
+  }
+  if (!texture || !texture->IsValid()) {
+    return false;
+  }
+
+  switch (stage) {
+    case ShaderStage::kVertex:
+      vertex_bindings.sampled_images.emplace_back(TextureAndSampler{
+          .slot = slot,
+          .texture = TextureResource::MakeDynamic(metadata, std::move(texture)),
+          .sampler = &sampler,
+      });
+      return true;
+    case ShaderStage::kFragment:
+      fragment_bindings.sampled_images.emplace_back(TextureAndSampler{
+          .slot = slot,
+          .texture = TextureResource::MakeDynamic(metadata, std::move(texture)),
+          .sampler = &sampler,
+      });
+      return true;
+    case ShaderStage::kCompute:
+      VALIDATION_LOG << "Use ComputeCommands for compute shader stages.";
+    case ShaderStage::kUnknown:
+      return false;
+  }
+
+  return false;
+}
```

Comment:
Please remove duplication with the above method.

Reply:
done